addons/mrp/models/stock_rule.py

```diff
@@ -45,38 +45,49 @@ def _should_auto_confirm_procurement_mo(self, p):
     def _run_manufacture(self, procurements):
         new_productions_values_by_company = defaultdict(list)
         for procurement, rule in procurements:
-            if float_compare(procurement.product_qty, 0, precision_rounding=procurement.product_uom.rounding) <= 0:
-                # If procurement contains negative quantity, don't create a MO that would be for a negative value.
-                continue
             bom = rule._get_matching_bom(procurement.product_id, procurement.company_id, procurement.values)
-
             mo = self.env['mrp.production']
-            mto_route = self.env['stock.warehouse']._find_global_route('stock.route_warehouse0_mto', _('Replenish on Order (MTO)'))
-            if rule.route_id != mto_route and procurement.origin != 'MPS':
-                gpo = rule.group_propagation_option
-                group = (gpo == 'fixed' and rule.group_id) or \
-                        (gpo == 'propagate' and 'group_id' in procurement.values and procurement.values['group_id']) or False
+            orderpoint = procurement.values.get('orderpoint_id')
+            gpo = rule.group_propagation_option
+            if (procurement.origin != 'MPS') and (orderpoint or gpo != 'propagate'):
```

Comment:
I would add a function that define if merge is allow or not. Could be useful for hooks

Reply:
done